src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py

```diff
@@ -357,55 +346,137 @@ def gui_submit(bundle_directory: str) -> Optional[dict[str, Any]]:
         return None
 
 
+def _options_dialog() -> dict[str, Any]:
+    """
+    Builds and displays a dialog within KeyShot to get the submission options
+    reuired before the main gui submission window is opened outside of KeyShot.
+    Options:
+        Option 1: Dropdown to select whether to submit just the scene file itself
+                  or all external file references as well by packing/unpacking a
+                  KSP bundle before submission.
+    Returns a dictionary of the selected option values in the format:
+        {'SUBMISSION_MODE_KEY': [1, 'only the scene BIP file']}
+    """
+    dialog_items = [
+        (
+            SUBMISSION_MODE_KEY,
+            lux.DIALOG_ITEM,
+            "Do you want to submit",
+            0,
+            ["the scene BIP file and all external files references", "only the scene BIP file"],
+        )
+    ]
+    selections = lux.getInputDialog(
+        title="AWS Deadline Cloud Submission Options",
+        values=dialog_items,
+        id=DEADLINE_CLOUD_DIALOG_ID,
+    )
+
+    return selections
+
+
+def _save_ksp_bundle(directory: str, bundle_name: str) -> str:
+    """
+    Saves out the current scene and any file references to a ksp bundle in a
+    directory.
+    Returns the file path where the bundle was saved to.
+    """
+    full_ksp_path = os.path.join(directory, bundle_name)
+
+    if not lux.savePackage(path=full_ksp_path):
+        raise RuntimeError("KSP was not able to be saved!")
+
+    return full_ksp_path
+
+
+def _get_ksp_bundle_files(directory: str) -> Tuple[str, list[str]]:
+    """
+    Creates a ksp bundle from the current scene containing the scene file and
+    any external file references. The bundle is unpacked into a directory passed
+    in.
+    Returns the scene file and a list of the external files from the directory
+    where the ksp was extracted to.
+    """
+
+    ksp_dir = os.path.join(directory, "ksp")
+    unpack_dir = os.path.join(directory, "unpack")
+
+    ksp_archive = _save_ksp_bundle(ksp_dir, "temp_deadline_cloud.ksp")
+    shutil.unpack_archive(ksp_archive, unpack_dir, "zip")
+    input_filenames = [
+        os.path.join(unpack_dir, file)
+        for file in os.listdir(unpack_dir)
+        if not file.endswith(".bip")
+    ]
+
+    bip_files = glob.glob(os.path.join(unpack_dir, "*.bip"))
+
+    if not bip_files:
+        raise RuntimeError("No .bip files found in the KSP bundle.")
```

Comment:
Should we raise an error if there are multiple bip files?

Reply:
I'm not sure that there is ever a case where they can be multiple BIP files, but I've added the error just in case.